### PR TITLE
StellarJS: workaround Stellar SDK bad error handling

### PR DESCRIPTION
**Pull request by ghost**
**How to reproduce**
On Android, at the end of an add accounts for Stellar, a 404 error message is systematically displayed.

**Root cause**
The Stellar SDK badly handles network answers from the underlying explorer (see https://github.com/stellar/js-stellar-sdk/issues/638), which makes it impossible for us to catch end of add accounts in a clean way.

**Fix**
Only dirty workarounds are possible at our level... we have no other way than to parse the errors the Stellar SDK returns, which are supposed to be cleanly typed but are actually completely different things depending on the cases...

**Things to test for QA**
- Add accounts on LLD with network connectivity
- Add accounts on LLD without network connectivity
- Add accounts on LLM with network connectivity
- Add accounts on LLM without network connectivity

### CODE REVIEW

**Comment by gre** [comment]
Is there a test that could assess the previous behavior is fixed? i'm not sure to understand exactly what to tell our QA regarding what was the problem.

File: src/families/stellar/api/horizon.js [hunk 1]
```diff
@@ -123,9 +128,22 @@ export const fetchOperations = async (
...
       return [];
     }
+    if (
+      e instanceof NetworkError ||
+      (e && e.toString().match(/Error: getaddrinfo ENOTFOUND/g)) ||
+      (e && e.toString().match(/undefined is not an object/g))
+    ) {
+      throw new NetworkDown();
+    }
+
```
Comment (rjarasson-ledger) [inline src/families/stellar/api/horizon.js:146]: I don't understand this part.
Can you add a helper to "remap errors" for the whole logic ?
It should be reused everywhere there is a catch all errors that could lead to network issues.

`.match(/status code 404/)` -> no throw (return logic after the remap)

`.match(/status code 4[0-9]{2}/)` => LedgerAPI4XX

`.match(/status code 5[0-9]{2}/)` => LedgerAPI5XX

`.match(/ECONNRESET|ECONNREFUSED|ENOTFOUND|EPIPE|ETIMEDOUT/)` => NetworkDown

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/families/stellar/api/horizon.js (hunk 1 above)
Comment (haammar-ledger) [reply src/families/stellar/api/horizon.js:146]: There's not always a status code, sometimes it's a raw JS `undefined` error coming from the Stellar SDK...

Comment (haammar-ledger) [reply src/families/stellar/api/horizon.js:146]: But I can replace this one `.match(/Error: getaddrinfo ENOTFOUND/g))` with the more detailed cases you mention.

Comment (rjarasson-ledger) [reply src/families/stellar/api/horizon.js]: yeah ok, so we can just add some cases. Strange behaviour though. Idea is just to handle as much (known) cases as possible.

**Review by ghost — APPROVED** [review]

**Comment by gre** [comment]
thanks for the comment; important tech debt on our side to track.

**Pull request merged**